src/modules/RootPart.rb

```diff
@@ -2165,15 +2165,29 @@ def staging
       Y2Storage::StorageManager.instance.staging
     end
 
-    def fstab_entry_matches?(entry, dev)
+    def fstab_entry_matches?(entry, filesystem)
       spec = entry["spec"]
-      return true if spec == dev
+      id, value = spec.include?("=") ? spec.split('=') : ["", spec]
 
-      # Why this doesn't match?
-      # Possible reasons:
-      # - /var not mounted so hwinfo cannot translate device names
-      Builtins.y2warning("Device does not match fstab: '%1' vs. '%2'", dev, spec)
-      return false
+      device =
+        if id.casecmp("LABEL") == 0
+          filesystem.label
+        elsif id.casecmp("UUID") == 0
+          filesystem.uuid
+        else
+          filesystem.blk_devices[0].name
+        end
+
+      matches = value == device
+
+      unless matches
```

Comment:
SUSE style guide, only use `unless` at the end of a sentence, not to open a block. That is, use either
```ruby
if !matches
  log.warning("bla")
end
```
or
```ruby
log.warning("bla") unless matches
```

If you ask me, I prefer the latter.